### PR TITLE
Some more cmake package fixes.

### PR DESCRIPTION
**Pull request by Hoikas**
This fixes two errors with the generated cmake packages:
- `HSPlasmaNet-targets.cmake` was not being exported , leading to some fun linker errors down the road.
- Absolute library paths were being exported, making sharing the built library a challenging prospect. I fixed all cases of absolute paths except for when using the optional feature PhysX.

Bumps the minimum CMake requirement to 3.12. (Welcome to the real world.)

### CODE REVIEW

File: net/CMakeLists.txt [hunk 1]
```diff
@@ -94,15 +94,11 @@ add_library(HSPlasmaNet
...
-endif()
-
-if(WIN32)
-    target_link_libraries(HSPlasmaNet crypt32)
-    target_link_libraries(HSPlasmaNet ws2_32)
-endif(WIN32)
+target_link_libraries(HSPlasmaNet
+                      HSPlasma string_theory
+                      $<$<NOT:$<PLATFORM_ID:SunOS>>:OpenSSL::Crypto>
+                      $<$<PLATFORM_ID:Windows>:ws2_32>
```
Comment (zrax) [inline net/CMakeLists.txt:100]: What happened to `crypt32`?  Is it not needed?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.12)
```
Comment (zrax) [inline CMakeLists.txt:1]: Huh, it looks like CI is using a newer CMake than what stock Ubuntu 18.04 ships...  Neat, I guess

File: net/CMakeLists.txt (hunk 1 above)
Comment (Hoikas) [reply net/CMakeLists.txt:100]: Correct, `OpenSSL::Crypto` correctly links all of OpenSSL's transitive dependencies. This is a major benefit to using targets, where available 😄

File: CMakeLists.txt (hunk 2 above)
Comment (Hoikas) [reply CMakeLists.txt:1]: Yeah, I think GHA runs the same CMake revisions across all of the hosted runners.

**Review by zrax — APPROVED** [review]

**Pull request merged**